### DISCUSSION TITLE
why route path doesn't support regexp

### DISCUSSION THREAD

**Discussion by caibirdme** [apache/apisix#6062] [Q&A] [answered]
I found that route path is either static path like /a/b or prefix match path like /a/*, I wonder why apisix does not support regexp directly and make it in proxy-rewrite plugin instead? And I want to know how to support REST API like blow:
1. /user/{id int}/profile  -> /a/b
2. /user/{id int}/order  -> /c/d

**Comment by tokers** [comment, marked as answer]
You can implement a regex pattern for a route, but you have to resort to the `vars` field, see https://github.com/api7/lua-resty-expr#new for details.

**Reply by caibirdme** [reply]
Thx, and could you please try to explain why apisix doesn't support regexp path directly? I mean, like in ngx, location support regexp, and in other api gateway like tyk traefic, regexp is also supported. So I want to know is there any specific consideration that apisix doesn't support it directly and make it a plugin instead

**Reply by tzssangglass** [reply]
APISIX's route match is based on https://github.com/api7/lua-resty-radixtree.

for more info, ref to: https://zhuanlan.zhihu.com/p/89610466
